commands/http/handler.go

```diff
@@ -71,6 +77,30 @@ func NewHandler(ctx cmds.Context, root *cmds.Command, allowedOrigin string) *Han
 	return &Handler{internal, c.Handler(internal)}
 }
 
+func NewReadOnlyHandler(ctx cmds.Context, root *cmds.Command, allowedOrigin string) *Handler {
```

Comment:
problem with making two handlers is drift in the implementations.

maybe instead of creating two different handlers, make the original handler always check a list of available commands, and then we can just instantiate the read-only and read-write handlers by passing two different command lists.

that way the 

```go
if i.readOnly == true {
  if _, ok := readOnlyCmds[req.Command()]; !ok { ... }
}
```

can turn into 

```go
if _, ok := availableCmds[req.Command()]; !ok { ... }
```